### ISSUE TITLE
No visible @interface for 'LifetimeConfiguration' declares the selector 'initWithMaxCount:groupName:'

### ISSUE THREAD

**Issue by ajfigueroa** [krzysztofzablocki/LifetimeTracker#34]
### Setup
- Xcode 9.2
- Mixed project of Swift 4 and Objective-C
- LifetimeTracker version 1.3.2 with a DEBUG configuration (I've also tried without this)
  - `pod 'LifetimeTracker', :configurations => ['Debug']`

### Issue

I'm trying to conform a class to `LifetimeTrackable` so I've followed the steps in the [README.md](https://github.com/krzysztofzablocki/LifetimeTracker#objective-c) for Objective-C integration.

I get the above error when I try to use `LifetimeConfiguration`: 
```
No visible @interface for 'LifetimeConfiguration' declares the
selector 'initWithMaxCount:groupName:'
```

In my file, I have imported the class as `@import LifetimeTrackable`.
I also tried importing as `#import <LifetimeTracker/LifetimeTracker-Swift.h>`. When I look at this file I see this generated interface for `LifetimeConfiguration`

```objective-c
/// Holds the properties which are needed to configure a <code>LifetimeTrackable</code>
SWIFT_CLASS("_TtC15LifetimeTracker21LifetimeConfiguration")
@interface LifetimeConfiguration : NSObject
- (nonnull instancetype)init SWIFT_UNAVAILABLE;
@end
```
Full generated interface file [here](https://gist.github.com/ajfigueroa/23e1842adec5555a385eb638af3e5f98)

Has this been seen before? I took a look at the example and I was unable to reproduce the issue.

**Comment by krzysztofzablocki** [comment]
hmm weird, the code works fine in our objc example, can you try adding @objc attribution to the initializer in your local pod checkout and see if it helps, I didn't expect this to be needed but maybe ?

**Comment by ajfigueroa** [comment]
@krzysztofzablocki Apologies for the delay! I was able to reproduce this more reliably now that I understand it better. Adding the @objc does indeed fix the issue.

The ExampleObjc project works fine because the pod itself is on Swift 3 as shown [here](https://github.com/krzysztofzablocki/LifetimeTracker/blob/master/ExampleObjC/Pods/Pods.xcodeproj/project.pbxproj#L452).
If you run the migrator so that it converts LifetimeTracker to Swift 4 which will require no syntax change then you'll see the following deprecation warning:

<img width="869" alt="screen shot 2018-02-14 at 2 07 47 pm" src="https://user-images.githubusercontent.com/3321592/36222854-9780db12-1190-11e8-95d2-e90e4bd7a506.png">

This matches what is happening in my project since we have a mixed Obj-C/Swift 4 codebase. I didn't run the migrator on the Pod when I installed it in my project as it was automatically set to Swift 4. 

I can make a PR to annotate the public methods to @objc or should it be done on a per need basis?
